docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,511 @@
+	if now := time.Now(); w.lastRead.Add(5 * time.Minute).Before(now) {
+		content, err := os.ReadFile(w.file)
+		if err != nil {
+			return "", err
+		}
+		w.assertion = string(content)
+		w.lastRead = now
+	}
+	return w.assertion, nil
+}
+
+```
+
+### <a name='OpenQuestions'></a>Open Questions
+
+#### <a name='Howtomultitenancy'></a>1. How to achieve multi-tenancy?
+
+The identity is tied to the client ID which can be supplied via AzureClusterIdentity.
+
+#### <a name='Howtodistributekeys'></a>2. How to distribute key pair to management cluster?
+
+Keys can be distributed by using the CABPK feature by creating a secret with name `<cluster-name>-sa` encompassing the key pair. More details on it [here](https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html)
+
+#### <a name='UserExperience'></a>3. User Experience
+
+Though AZWI has a lot of advantages as compared to AAD pod identity, setting up AZWI involves couple of manual step and it can impact the user experience.
+
+### <a name='MigrationPlan'></a>Migration Plan
+
+Management clusters using AAD pod identity should have a seamless migration process which is well documented.
+
+For migrating an existing cluster to use AZWI following steps should be taken.
+
+**Migrating Greenfield Clusters**
+- The steps in this section applies to greenfield cluster creation that uses CAPI pivoting to use workload identity. As in this case the key pairs are already distributed to the control plane nodes.
+
+- Create a `UserAssignedIdentity` and create a new machine template to use user assigned identity. This step will become optional once cloud provider azure supports workload identity.
+
+- Patch the control plane to include the following flag on the api server.
+	- `service-account-issuer: <value-is-service-account-issuer-url>
+
+- Perform node rollout to propagate the changes by patching KCP and MachineDeployment objects to include the reference on the new AzureMachineTemplate created in the previous step.
+
+- Create a new `AzureClusterIdentity` and specify the client ID, tenant ID and `type` to `WorkloadIdentity`.
+
+- Patch AzureCluster to use the new `AzureClusterIdentity`.
+
+**Migrating Brownfield Clusters**
+
+- The steps in this section applies to existings clusters. 
+
+- Generate a key pair or use the same key pairs as present in `/etc/kubernetes/pki`
+
+- Perform the following pre-requistes as discussed earlier in the **Bootstrap Cluster** section of the [document](#proposal):
+	- Generate and upload the JWKS and Discovery Document.
+	- Install the AZWI mutating admission webhook controller.
+	- Establish federated credential for the key pair.
+
+- If you are not using the existing key pairs then perform a service account rotation using this guide. 
+https://azure.github.io/azure-workload-identity/docs/topics/self-managed-clusters/service-account-key-rotation.html
+
+- If you are using the existing key pairs then rotation is not required.
```

Comment:
Or if you just set configure workload identity using the keys present in `/etc/kubernetes/pki`